src/components/dataTable/js/tableDirective.js

```diff
@@ -0,0 +1,131 @@
+angular
+  .module('material.components.table')
+  .directive('mdTable', MdTableDirective);
+
+/**
+ * @ngdoc directive
+ * @name mdTable
+ * @module material.components.dataTable
+ *
+ * @description
+ *
+ * Data tables allow users to view and manipulate large data sets efficiently.
+ *
+ * Data tables are composed of a component hierarchy, with the `md-table` component being the root
+ * component. An important distinction between this component and other components is data tables
+ * **do not** use custom elements. The reason data tables do not use custom elements is to preserve
+ * functionality of HTML tables. For example, `rowspan` and `colspan` do not work with CSS tables.
+ *
+ * @usage
+ *
+ * <hljs lang="html">
+ * <table md-table>
+ *   <tbody>
+ *     <tr>
+ *       <td></td>
+ *     </tr>
+ *   </tbody>
+ * </table>
+ * </hljs>
+ *
+ * @param {boolean=} md-row-select Enables row selection.
+ * @param {*=} md-selected A model for the selected item(s). If multiple selection is enabled the selected items will be stored in an array.
+ * @param {boolean=} multiple Allows multiple items to be selected.
+ */
+function MdTableDirective() {
+
+  return {
+    restrict: 'A',
+    bindToController: true,
+    scope: {
+      selected: '=?mdSelected'
+    },
+    compile: compileFn,
+    controller: MdTableController,
+    controllerAs: '$mdTable'
+  };
+
+  function compileFn(tElement) {
+    tElement.addClass('md-table');
+  }
+}
+
+function MdTableController($attrs, $element, $scope, $mdUtil) {
+  this.$element = $element;
+  this.$attrs = $attrs;
+  this.$scope = $scope;
+  this.$mdUtil = $mdUtil;
+
+  this.unregisterWatcher = null;
+  this.changeListeners = [];
+
+  var self = this;
+
+  $attrs.$observe('multiple', function (multiple) {
+    self.multiple = $mdUtil.parseAttributeBoolean(multiple);
+
+    if (self.multiple) {
+      // This transforms our previous selected value in a array for the multiple selections.
+      self.selected = [].concat(self.selected || []);
```

Comment:
it could be `undefined`, in which case `self.selected := [undefined]` and `self.selected.length := 1` even though none of the items are selected.